test/common/common/radix_tree_test.cc

```diff
@@ -0,0 +1,356 @@
+  EXPECT_TRUE(radixtree.add(key_a, cstr_a));
+  EXPECT_EQ(cstr_a, radixtree.find(key_a));
+}
+
+TEST(RadixTree, RadixTreeSpecificTests) {
+  RadixTree<const char*> radixtree;
+  const char* cstr_a = "a";
+  const char* cstr_b = "b";
+  const char* cstr_c = "c";
+  const char* cstr_d = "d";
+
+  // Test radix tree compression
+  EXPECT_TRUE(radixtree.add(std::string("test"), cstr_a));
+  EXPECT_TRUE(radixtree.add(std::string("testing"), cstr_b));
+  EXPECT_TRUE(radixtree.add(std::string("tester"), cstr_c));
+  EXPECT_TRUE(radixtree.add(std::string("tested"), cstr_d));
+
+  EXPECT_EQ(cstr_a, radixtree.find("test"));
+  EXPECT_EQ(cstr_b, radixtree.find("testing"));
+  EXPECT_EQ(cstr_c, radixtree.find("tester"));
+  EXPECT_EQ(cstr_d, radixtree.find("tested"));
+
+  // Test prefix matching
+  EXPECT_THAT(radixtree.findMatchingPrefixes("test"), ElementsAre(cstr_a));
+  EXPECT_THAT(radixtree.findMatchingPrefixes("testing"), ElementsAre(cstr_a, cstr_b));
+  EXPECT_THAT(radixtree.findMatchingPrefixes("tester"), ElementsAre(cstr_a, cstr_c));
+  EXPECT_THAT(radixtree.findMatchingPrefixes("tested"), ElementsAre(cstr_a, cstr_d));
+
+  // Test longest prefix
+  EXPECT_EQ(cstr_a, radixtree.findLongestPrefix("test"));
+  EXPECT_EQ(cstr_b, radixtree.findLongestPrefix("testing"));
+  EXPECT_EQ(cstr_c, radixtree.findLongestPrefix("tester"));
+  EXPECT_EQ(cstr_d, radixtree.findLongestPrefix("tested"));
+  EXPECT_EQ(cstr_a, radixtree.findLongestPrefix("testx"));
+  EXPECT_EQ(nullptr, radixtree.findLongestPrefix("tex"));
+}
+
+TEST(RadixTree, EmptyAndSingleNode) {
+  RadixTree<const char*> radixtree;
+  const char* cstr_a = "a";
+
+  // Test empty radixtree
+  EXPECT_EQ(nullptr, radixtree.find("anything"));
+  EXPECT_EQ(nullptr, radixtree.findLongestPrefix("anything"));
+  EXPECT_THAT(radixtree.findMatchingPrefixes("anything"), ElementsAre());
+
+  // Test single node
+  EXPECT_TRUE(radixtree.add(std::string("a"), cstr_a));
+  EXPECT_EQ(cstr_a, radixtree.find("a"));
+  EXPECT_EQ(cstr_a, radixtree.findLongestPrefix("a"));
+  EXPECT_THAT(radixtree.findMatchingPrefixes("a"), ElementsAre(cstr_a));
+  EXPECT_EQ(nullptr, radixtree.find("b"));
+  EXPECT_EQ(nullptr, radixtree.findLongestPrefix("b"));
+  EXPECT_THAT(radixtree.findMatchingPrefixes("b"), ElementsAre());
+}
+
+TEST(RadixTree, InsertAndFindEdgeCases) {
+  RadixTree<const char*> radixtree;
+  const char* cstr_a = "a";
+  const char* cstr_b = "b";
+  const char* cstr_c = "c";
+  const char* cstr_d = "d";
+
+  // Test empty string
+  EXPECT_TRUE(radixtree.add(std::string(""), cstr_a));
+  EXPECT_EQ(cstr_a, radixtree.find(""));
+  EXPECT_EQ(cstr_a, radixtree.findLongestPrefix(""));
+  EXPECT_THAT(radixtree.findMatchingPrefixes(""), ElementsAre(cstr_a));
+
+  // Test single character
+  EXPECT_TRUE(radixtree.add(std::string("x"), cstr_b));
+  EXPECT_EQ(cstr_b, radixtree.find("x"));
+  EXPECT_EQ(cstr_b, radixtree.findLongestPrefix("x"));
+  EXPECT_THAT(radixtree.findMatchingPrefixes("x"), ElementsAre(cstr_b));
```

Comment:
This test case is failing to catch the incorrect behavior already mentioned in another comment - the empty string *is* a matching prefix in the tree at this point, so it should be `ElementsAre(cstr_a, cstr_b)`.